privaterelay/plans.py

```diff
@@ -0,0 +1,610 @@
+get_phone_country_language_mapping, and get_bundle_country_language_mapping,
+all which return a PlanCountryLangMapping dict, which has this structure:
+
+{
+  "at": {
+    "de": {
+      "monthly": {
+        "id": "price_1LYC79JNcmPzuWtRU7Q238yL",
+        "price": 1.99,
+        "currency": "EUR",
+      },
+      "yearly": {
+        "id": "price_1LYC7xJNcmPzuWtRcdKXCVZp",
+        "price": 0.99,
+        "currency": "EUR",
+      },
+    },
+  },
+  ...
+}
+
+This says that Austria (RelayCountryStr "at") defaults to German (LanguageStr "de"),
+and has a monthly and a yearly plan. The monthly plan has a Stripe ID of
+"price_1LYC79JNcmPzuWtRU7Q238yL", and costs €1.99 (CurrencyStr "EUR"). The yearly
+plan has a Stripe ID of "price_1LYC7xJNcmPzuWtRcdKXCVZp", and costs €11.88 a year,
+equivalent to €0.99 a month.
+
+The top-level keys say which countries are supported. The function get_premium_countries
+returns these as a set, when the rest of the data is unneeded.
+
+The second-level keys are the languages for that country. When the country is known but
+the language is not, or is not one of the listed languages, the first language is the
+default for that country.
+
+The third-level keys are the plan periods. Premium and phones are available on
+monthly and yearly periods, and bundle is yearly only.
+
+The raw data is stored in two dicts:
+* _STRIPE_PLAN_DATA
+* _RELAY_PLANS_BY_COUNTRY_AND_LANGUAGE
+
+These are extended to support more countries, languages, plans, etc. They are parsed
+on first use to create the PlanCountryLangMapping, and served from cache on later uses.
+"""
+
+from copy import deepcopy
+from functools import lru_cache
+from typing import Literal, TypedDict
+
+from django.conf import settings
+
+#
+# Public functions
+#
+
+
+def get_premium_country_language_mapping(
+    eu_country_expansion: bool | None,
+) -> "PlanCountryLangMapping":
+    """Get mapping for premium countries (unlimited masks, custom subdomain)"""
+    return _country_language_mapping(
+        "premium", eu_country_expansion=eu_country_expansion
+    )
+
+
+def get_premium_countries(eu_country_expansion: bool | None) -> set["RelayCountryStr"]:
+    """Get the country codes where Relay premium can be sold."""
+    mapping = get_premium_country_language_mapping(eu_country_expansion)
+    return set(mapping.keys())
+
+
+def get_phone_country_language_mapping() -> "PlanCountryLangMapping":
+    """Get mapping for phone countries (premium + phone mask)"""
+    return _country_language_mapping("phones")
+
+
+def get_bundle_country_language_mapping() -> "PlanCountryLangMapping":
+    """Get mapping for bundle countries (premium + phone mask + VPN)"""
+    return _country_language_mapping("bundle")
+
+
+#
+# Data
+#
+
+# Selected Stripe data
+# The "source of truth" is the Stripe data, this copy is used for upsell views
+# and directing users to the correct Stripe purchase page.
+_STRIPE_PLAN_DATA: "_StripePlanData" = {
```

Comment:
question (blocking): why are we using strings for some type annotations instead of symbols? I don't see anything in these type definitions that need run-time evaluation. can we move the type declarations up above the code that references them so we can use symbols instead of strings?